core/src/main/java/org/jdbi/v3/core/mapper/reflect/ReflectionMappers.java

```diff
@@ -60,21 +58,25 @@ private ReflectionMappers(ReflectionMappers that) {
     }
 
     /**
-     * Returns the registered column name mappers.
+     * Returns the registered column name mappers in a new list
+     * that may be iterated and modified by the caller.
      *
      * @return the registered column name mappers.
      */
     public List<ColumnNameMatcher> getColumnNameMatchers() {
-        return Collections.unmodifiableList(columnNameMatchers);
```

Comment:
the unmodifiable list is IMHO the better change because it makes it impossible for the caller to modify the returned list.

Reply:
I left two comments on the original ticket #2238 to explain.

Reply:
They can not. For concurrent modification exception, it would need to be possible to modify the internal list object. You can not do that.

Reply:
You're right.